### DISCUSSION TITLE
get current datatable filters state

### DISCUSSION THREAD

**Discussion by Amit-ga** [primefaces/community#1366] [PrimeReact] [answered]
Hello,
I am trying to allow the user data filtering  via URL in primereact datatable.
I am working on a URL generator button that shows a popup with the URL that matches the current selected filters.
How can i get the current selected filters from the table at any given time? after clicking apply in a filter menu, where does it get updated?

Thank you.

**Comment by Amit-ga** [comment]
still hasnt been able to make this  work, my filters variable just does not get updated when clicking apply on a filter menu.
Would love your help with this! :)

**Reply by melloware** [reply]
@Amit-ga i think this needs to be requested as an enhancement to expose the current filters on the DT ref like `dtRef.current?.getFilters()` ???

**Reply by Amit-ga** [reply]
Thank you for your answer.
I am new to this, how does requesting an enhancement work? I see my question is classified as an enhancement, is there anything I need to do on my side?
How long does getting this enhancement usually  take? will I get updates about my request?

Thank you!

**Reply by melloware** [reply]
https://github.com/primefaces/primereact/issues

**Reply by Amit-ga** [reply]
Hello, I was wondering if there is any progress with this?

Thanks

**Reply by melloware** [reply]
No it doesn't look like you ever opened a GitHub issue ticket so why would there be any progress?

**Comment by melloware** [comment, marked as answer]
Fixed and in 10.6.5 if you want to try it.

**Reply by Amit-ga** [reply]
thank you so much! 
Can you please write an example of usage so i know the proper syntax or direct me to where i cam find it?

**Reply by melloware** [reply]
Yep there are two more methods on the `ref`.

https://primereact.org/datatable/#api.DataTable.methods.getFilterMeta

https://primereact.org/datatable/#api.DataTable.methods.setFilterMeta

So `datatableRef?.current.getFilterMeta()` to get the current filters.
